### PR TITLE
📦  Refactor to use `@citation-js/core`

### PR DESCRIPTION
**Pull request by agoose77**
This PR replaces the use of `citation-js` with `@citation-js/core`. With the CSL and bibtex plugins, the installed size is <1 MB.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,5 +33,9 @@
     "npm": ">=7.0.0",
     "node": ">=14.0.0"
   },
-  "packageManager": "npm@8.10.0"
+  "packageManager": "npm@8.10.0",
+  "dependencies": {
+    "@citation-js/core": "^0.6.8",
+    "@citation-js/plugin-csl": "^0.6.9"
+  }
```
Comment (rowanc1) [inline package.json:40]: We should only have to add these in the sub-package, not at the top!

**Review by rowanc1 — CHANGES_REQUESTED** [review]
Nice, this looks good - I will do a quick test later today and then I think we can bring it in.

- [ ] fix up package.json in root
- [x] add changeset for citation-js-utils

File: package.json (hunk 1 above)
Comment (agoose77) [reply package.json:40]: Wait, how did this happen?! I think my editor must have done this. I didn't notice the path when I was being lazy and using update for the add strategy. Yikes!

**Comment by agoose77** [comment]
@rowanc1 I've run `prettier` in each package, and its changed some packages that I've not touched in this PR. Is there some kind of version dependency in prettier that's not captured by our development dependencies? It's interesting that the CI seems happy (i.e. my linting is not contra whatever version of prettier other people are running).

**Comment by rowanc1** [comment]
Looks like the `package-lock` change is quite extensive, and changed prettier from `v3.0.0` --> `v3.2.4`, which is the root of these differences.

**Comment by agoose77** [comment]
Ah, I was thinking about this the wrong way then. Fab, fixed.

**Comment by rowanc1** [comment]
That is awesome, thanks for making the lock-diff much smaller!

Could you add a changeset for each package changed:

```
npm run changeset
```

and follow the instructions. This allows us to keep track of which packages need to be bumped and released next time we version.

**Comment by agoose77** [comment]
> That is awesome, thanks for making the lock-diff much smaller!
> 
> Could you add a changeset for each package changed:
> 
> ```
> npm run changeset
> ```
> 
> and follow the instructions. This allows us to keep track of which packages need to be bumped and released next time we version.

Yes, sorry — been meaning to do that. Not sure why it slipped my mind to begin with :upside_down_face:

**Review by rowanc1 — APPROVED** [review]
Fell into the same problems, I put back in those changes and we can update the linting and merge it in!

File: package-lock.json [hunk 2]
```diff
@@ -9768,9 +9619,9 @@
       }
     },
     "node_modules/prettier": {
-      "version": "3.0.0",
-      "resolved": "https://registry.npmjs.org/prettier/-/prettier-3.0.0.tgz",
-      "integrity": "sha512-zBf5eHpwHOGPC47h0zrPyNn+eAEIdEzfywMoYn2XPi0P44Zp0tSq64rq0xAREh4auw2cJZHo9QUob+NqCQky4g==",
+      "version": "3.2.4",
```
Comment (rowanc1) [inline package-lock.json:9622]: This is the root of the format changes.

File: package-lock.json [hunk 3]
```diff
@@ -13357,7 +13188,9 @@
       "version": "1.0.2",
       "license": "MIT",
       "dependencies": {
-        "citation-js": "^0.6.3",
+        "@citation-js/core": "^0.7.6",
+        "@citation-js/plugin-bibtex": "^0.7.8",
+        "@citation-js/plugin-csl": "^0.7.6",
```
Comment (rowanc1) [inline package-lock.json:13193]: @agoose77 I went to the next version, seems fine in testing!

**Pull request merged**